kommunicate/src/main/java/com/applozic/mobicomkit/broadcast/BroadcastService.java

```diff
@@ -190,6 +181,26 @@ public static void sendNotificationBroadcast(Context context, Message message, i
         }
     }
 
+    //Handle Routing rules for Agents
+    //Notify everybody -> Send notification to all agent, along with assignment message
+    //Automatic assignment -> Send notification only to the assigned agent
+    private static boolean showNotificationToAgent(Context context, Contact contact, Channel channel, Message message) {
+        MobiComUserPreference userPreference = MobiComUserPreference.getInstance(context);
+        if (userPreference.getUserRoleType() == 3 || channel == null || contact == null) {
+            return false;
+        }
+        if (userPreference.isNotifyEverybody()) {
+            if (User.RoleType.BOT.getValue().equals(new AppContactService(context).getContactById(channel.getConversationAssignee()).getRoleType())) {
+                return (message.getMetadata() != null && message.getMetadata().containsKey(Message.BOT_ASSIGN));
```

Comment:
@amntoppo this block is handling which case?

Reply:
@adarshmishra This handles the Assignment message notification -   "Assigned to Aman"

Reply:
Without the assignee role check, it shows notification even when bot is handling the conversation.

Reply:
Is it similar to web?

Reply:
Yes, the flow is same as web now.